### DISCUSSION TITLE
Moving to Shiny 3.0 with a classic/non-MAUI setup

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1395] [Q&A]
I can see that IShinyStartup has gone and can see the breaking changes listed here, https://shinylib.net/release-notes/client/v30/

> IShinyStartup is gone - we now have a new HostBuilder pattern to allow classic to align with MAUI/NET6. Use extensions methods to build on this for classic/non-MAUI setups

Perhaps I am being stupid (again) but do you have example of using the HostBuilder pattern in Xamarin.Forms?

**Comment by munkii** [comment]
Ignore for now. I think I have understoos how to do it (embarassingly). Assuming it works I will close this with a "comment" that shows what i did to get it working

**Comment by munkii** [comment]
Simply closing this as adding any samples or code  would help no one it is that simple

**Reply by aritchie** [reply]
You're pretty hard on yourself.  No need to be.  I'll see if I can add a bit more to the manual setup that includes classic xamarin stuff.  The hostbuilder setup was meant to look partially like the MauiProgram kind of setup.
